### ISSUE TITLE
Allow bulk add/register users

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#1823]
Add an option to the add users page for individual books that allows a user with permission to add users to the book to add many users at once. The admin should see a role selection dropdown and an input form: a text entry field that allows them to enter multiple email address (one per line). 

If a user with one of the email addresses already exists in the network, that user should be added to the book. If one does not exist, a new user invitation should be sent, using the text string that precedes the `@` sign for the user name. If a username already exists for the string or the string is not 4 characters long, extra characters should be added for disambiguation/padding. Following the completion of the activity, some kind of status message should be shown to the user who added the new users.

**Comment by SteelWagstaff** [comment]
Spec further: add network manager option (with URL) and add simpler option on book add user page.

**Comment by mcgratay** [comment]
The important bits: 
- only the correct roles have access to this tool and only when the correct features have been enabled 
- it works when it's supposed to work
- it blocks who it's supposed to block 

Issues that I'd fix now: 
- No emails are sent to existing users to let them know they have been added to a new book. For context, the default add user form lets you choose to or not to send a confirmation email. One solution is to add the "skip confirmation email" checkbox to this page. Another is to force confirmation email for all existing and new users. 
- the banner reads "user was successfully added" when **new** users are added by bulk, but those users **aren't** successfully added yet and may never be. They have to activate their account first. The default add user form has this banner instead when a new user account is being created and added to the book simultaneously: 

> Invitation email sent to new user. A confirmation link must be clicked before their account is created. 

I'd recommend making that the behavior here too. There are cases where both banners should appear at once. 

Things I'd change eventually: 
- There's no reason to have both the default Add New Users page and the Bulk Add page. The Bulk Add page performs all of the actions that you can do on the Add New Users one-at-a-time page. I'd suggest eventually replacing the default page with the Bulk Add page eventually, with the following in mind: 
  - The title "Bulk Add" can just become "Add Users" 
  - If the network level setting "Allow book administrators to invite new users as collaborators on their book" is **not** enabled, then book admins can only add existing users to their book. Currently, that removes the Bulk Add page completely, but keeps the "Add Existing Users" form on the default Add User page.

**Comment by cagp-dev-mtl** [comment]
After our discussion with @mcgratay and @SteelWagstaff, we are applying the following recommendation:

![Screenshot_2019-11-20_09-11-48](https://user-images.githubusercontent.com/8563418/69245951-d0c6f380-0b75-11ea-83d1-2f99023bac54.png)

**PR for this change is in review:**
[Pb 1823 bulk add users #1838](https://github.com/pressbooks/pressbooks/pull/1838)

**Comment by mcgratay** [comment]
I believe the system is doing the right thing but displaying the wrong thing. See this image: 

![Screen Shot 2019-11-25 at 3 16 57 PM](https://user-images.githubusercontent.com/39064028/69575250-d9656280-0f97-11ea-90aa-0ec1897acaa9.png)


If you look at the flags at the top you'll see that taylor+test5234 was added successfully to the book, but that taylor@pressbooks needs to confirm their account. It's actually the opposite - taylor@pressbooks is an existing user as well as a network admin, and taylor+test5234 needs to confirm their account. 

Even though the flag states that taylor+test5234 has successfully been added, I did receive a confirmation email to complete registration for the account. Even though the flag states taylor@pressbooks needs to confirm their account, that account was added to the user list immediately.
